### PR TITLE
feat: echidna and halmos testing campaign

### PR DESCRIPTION
**Pull request by wei3erHase**
migrating `test/invariant` folder from #71.

### CODE REVIEW

File: test/invariant/PROPERTIES.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+| [yes](test/unit/BFactory.t.sol:139) | BFactory's blab should always be modifiable by the current blabs                                            |
+| [yes](test/unit/BFactory.t.sol:198) | BFactory should always be able to transfer the BToken to the blab, if called by it                          |
+| [yes](test/unit/BPool.t.sol:1553)   | the amount received can never be less than min amount out                                                   |
+| [yes](test/unit/BPool.t.sol:1871)   | the amount spent can never be greater than max amount in                                                    |
+
+## Variable transition
+
+| Tested                           | Property                                                                                                    |
+| -                                | ----------------------------------------------------------------------------------------------------------- |
+| [yes](test/unit/BToken.t.sol:38) | BToken increaseApproval should increase the approval of the address by the amount                           |
```
Comment (0xteddybear) [inline test/invariant/PROPERTIES.md:17]: @drgorillamd is this kind of testing we're performing with foundry good enough? should we cover it with echidna as well?

Comment (drgorillamd) [reply test/invariant/PROPERTIES.md:17]: Nope, that's the idea: fuzzing one param gives a ok-answer (small fuzzing space + stateless), fuzzing campaign/echidna gives a better answer (bigger space + statefull), symb exec gives a math certainty (whole input space is covered, but stateless only)

Comment (drgorillamd) [reply test/invariant/PROPERTIES.md:17]: all in all, there is no perfect test, demonstrating safe and soundness is impossible, *but* we reduce the hole size in every cheese layer, then combine layers hoping no holes align

Comment (drgorillamd) [reply test/invariant/PROPERTIES.md:17]: image

Comment (0xteddybear) [reply test/invariant/PROPERTIES.md:17]: thanks for the explaination :heart:

**Pull request closed without merging**